msgpack/_unpacker.pyx

```diff
@@ -20,6 +19,7 @@ from cpython.exc cimport PyErr_WarnEx
 cdef extern from "Python.h":
     ctypedef struct PyObject
     cdef int PyObject_AsReadBuffer(object o, const void** buff, Py_ssize_t* buf_len) except -1
+    cdef int PyObject_GetBuffer(object o, Py_buffer *view, int flags) except -1
```

Comment:
You are right. I reverted this change and it had no effect.
At first I could not reproduce the problem, but I think the control flow was wrong, because Cython does not automatically exit a ``cdef`` function when exceptions occur. Checking for errors and explicitly re-raising does the job.